Source/Request.swift

```diff
@@ -535,6 +578,30 @@ open class Request {
 
         return self
     }
+
+    // MARK: - Response Serializers
+
+    /// Appends the response serialization closure to the `Request`.
+    ///
+    /// - Parameter responseSerializer: The responser serialization closure.
+    open func appendResponseSerializer(_ closure: @escaping () -> Void) {
+        protectedMutableState.write { $0.responseSerializers.append(closure) }
+    }
+
+    /// Notifies the `Request` that the response serializer is complete.
+    ///
+    /// - Parameter completion: The closure to execute once all response serializers are executed.
+    open func responseSerializerDidComplete(completion: @escaping () -> Void) {
+        protectedMutableState.write { $0.responseSerializerCompletions.append(completion) }
+        processNextResponseSerializer()
+    }
```

Comment:
We could probably get away with making these APIs `internal` if we really wanted to hide all this functionality.

Reply:
I think we can make a final decision once the APIs are finalized.

Reply:
Moved to `internal` APIs in 4dea9036 since users should always go through the `response` APIs.